doc/rados/configuration/ceph-conf.rst

```diff
@@ -4,91 +4,92 @@
  Configuring Ceph
 ==================
 
-When Ceph services start, the initialization process activates a series
-of daemons that run in the background. A :term:`Ceph Storage Cluster` runs 
-at a minimum three types of daemons:
+When Ceph services start, the initialization process activates a series of
+daemons that run in the background. A :term:`Ceph Storage Cluster` runs at
+least three types of daemons:
 
 - :term:`Ceph Monitor` (``ceph-mon``)
 - :term:`Ceph Manager` (``ceph-mgr``)
 - :term:`Ceph OSD Daemon` (``ceph-osd``)
 
 Ceph Storage Clusters that support the :term:`Ceph File System` also run at
-least one :term:`Ceph Metadata Server` (``ceph-mds``). Clusters that
-support :term:`Ceph Object Storage` run Ceph RADOS Gateway daemons
-(``radosgw``) as well.
+least one :term:`Ceph Metadata Server` (``ceph-mds``). Clusters that support
+:term:`Ceph Object Storage` run Ceph RADOS Gateway daemons (``radosgw``).
 
-Each daemon has a number of configuration options, each of which has a
-default value.  You may adjust the behavior of the system by changing these
-configuration options.  Be careful to understand the consequences before
+Each daemon has a number of configuration options, each of which has a default
+value. You may adjust the behavior of the system by changing these
+configuration options. Be careful to understand the consequences before
 overriding default values, as it is possible to significantly degrade the
-performance and stability of your cluster.  Also note that default values
-sometimes change between releases, so it is best to review the version of
-this documentation that aligns with your Ceph release.
+performance and stability of your cluster. Note too that default values
+sometimes change between releases. For this reason, it is best to review the
+version of this documentation that applies to your Ceph release.
 
 Option names
 ============
 
-All Ceph configuration options have a unique name consisting of words
-formed with lower-case characters and connected with underscore
-(``_``) characters.
+Each of the Ceph configuration options has a unique name that consists of words
+formed with lowercase characters and connected with underscore characters
+(``_``).
```

Comment:
Noticing this one way too late so please excuse me.

I would just want to comment that this is not strictly speaking true. There are configuration option names such as `mgr/cephadm/device_enhanced_scan` that contain slashes. They seem to have some kind of structure and it would be nice to either explain this, or at the minimum not act like it doesn't exist and imply that slashes are not valid in option names.

I admit I may be missing some nuance about these "slash option names" but that's because they are not explained anywhere except maybe the code where I didn't look as of yet.

Reply:
I think this was written before cephadm was a thing.  But in your specific case ...

`doc/cephadm/services/osd.rst` contains `ceph config set mgr mgr/cephadm/device_enhanced_scan true`.  Is your example an option name, or a central config mask?

From teh_src

```    // try to parse the variable name into var, either \$\{(.+)\} or
    // \$[a-z\_]+
    const char *valid_chars = "abcdefghijklmnopqrstuvwxyz_";```

That said, I'm not immediately finding where spaces within option names are translated.

Reply:
> Is your example an option name, or a central config mask?

I do not know what they are, I do not know what is a "central config mask". I am aware that `who` parameter (`mgr` in the above example) can have slashes and that *is* explained in the relevant doc.  
To my knowledge this particular syntax has not been explained anywhere? And if it is explained somewhere, perhaps it should be moved here or at the minimum be linked from this paragraph?

It's not only `cephadm` there is also dashboard-related options like this (`ceph config dump` output):
```
mgr                                advanced  mgr/dashboard/AUDIT_API_ENABLED         true                                                                             
```

Whatever it is, I'd hope it would have same kind of explanation as the slashes for the `who` parameter of `ceph config set` in this doc.

It clearly is not optional since just `ceph config set mgr device_enhanced_scan` will not work.

I guess my main point is that claiming that the valid charset does not include slashes will just confuse people even more when they see these kind of config options.

Reply:
After looking at the document, section `Sections and masks`: I believe this is the "central config mask" you refer to? But if I am reading correctly this applies to the `who` of the config option, and not to the config option name?

Specifically it says:
> (in the following examples, this is the "who" string) may be a section name, a mask, or a combination of both separated by a slash character (``/``)

And in this example `mgr` is the "who" listed in the quote, the slashes are not on it - they are in the configuration option name instead?